docs/openapi.yaml

```diff
@@ -976,6 +976,15 @@ paths:
       tags:
         - Smart Contracts
       operationId: get_contracts_by_trait
+      requestBody:
+        description: Trait Application Binary Interface(ABI) in the JSON format
+        required: true
+        content:
+          application/json:
+            schema:
+              type: string
+            example: 
+              $ref: ./entities/contracts/contracts-by-trait.example.json
```

Comment:
I'm not sure if this is correct. The endpoint is `GET /extended/v1/contract/by_trait?trait_abi=<value>` where `<value>` is the [URI encoded](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) JSON-stringified ABI object.

So rather than `requestBody` here (which I think is mostly for POST requests), seems like it should be something like:
```
parameters:
  - in: query
    name: by_trait
    content:
      application/json:
        schema:
          type: object
          properties: ...
```

This suggestion was just from a cursory look over how to specify json in a query param -- you should double check if this is actually the expected approach.

Reply:
If OpenAPI schema doesn't actually allow specifying a JSON-stringified query param, then you may just have to specify a string query param and document that it needs to be the JSON-stringified object, e.g.:
```
parameters:
  - in: query
    name: by_trait
    description: JSON stringified ABI object e.g. `JSON.stringify(abi)`
    schema:
      type: string
```

Reply:
I tried to add the following code to the yaml file, but I don't see any change in the output. May I know if this is the right way to add parameters to the OpenAPI.yaml?
```
parameters:
        - name: trait_abi
          in: query
          description: JSON stringified ABI object e.g. `JSON.stringify(abi)`.
          required: true
           schema:  # The request schema
              type: object  # JSON object type
              properties:
                functions:
                  type: array
                  items:
                    type: object
                    properties:
                      access:
                        type: string
                      args:
                        type: array
                        items: {}
                      name:
                        type: string
                      outputs:
                        type: object
                        properties:
                          type:
                            type: object
                            properties:
                              response:
                                type: object
                                properties:
                                  ok:
                                    type: string
                                    format: uint128
                                  error:
                                    type: string
                                    format: uint128
                variables:
                  type: array
                  items: {}
                maps:
                  type: array
                  items: {}
                fungible_tokens:
                  type: array
                  items: {}
                non_fungible_tokens:
                  type: array
                  items: {} ```

Reply:
If the above approach is incorrect, I will go ahead with [this](https://github.com/hirosystems/stacks-blockchain-api/pull/1699/commits/76f53d824c1b58f9b533061721365fb984dd36cc) commit.

Reply:
The OpenAPI 3.0 spec on object parameters is here: https://swagger.io/specification/v3/#parameter-object-examples

It looks like the first example I gave may be correct. I don't think you should be trying to define the entire possible ABI object here though, you're going to end up having to specify a large complex enum of clarity types. I seems fine to me if the spec simple states a `string` and just explain the JSON encoded abi in the description.